lib/base-compiler.js

```diff
@@ -328,70 +328,62 @@ class BaseCompiler {
         return this.runCompiler(compiler, options, inputFilename, execOptions);
     }
 
-    buildExecutableInFolder(key, dirPath) {
+    async buildExecutableInFolder(key, dirPath) {
```

Comment:
I love how clean it all looks

Reply:
I can't wait to bust outr async in the main compile method